tasks/install-plugins.yml

```
@@ -37,16 +37,26 @@
         {{ jenkins_pimt_args }}
       environment:
         CACHE_DIR: "{{ jenkins_caches_dir }}"
+      # Weirdly, no_log wants to interpret `jenkins_pimt_logfile != ''` as a string, not
+      # a boolean. So we need to use the ternary filter again.
+      no_log: "{{ (jenkins_pimt_logfile != '') | ternary(true, false) }}"
```

```suggestion
      # Weirdly, no_log wants to interpret `jenkins_pimt_logfile != ''` as a string, not
      # a boolean.
      no_log: "{{ (jenkins_pimt_logfile != '') | bool }}"
```
Maybe the comment is not necessary.

---

Well, it's not, but only because the filter doesn't work, either. 😅 I tried both the `ternary` and the `bool` filters, and actually, in all cases, `no_log` _really_ wants to interpret them as strings. I'm not sure why this is and I'll file a bug on Ansible's GitHub repo when I get around to it.

In the meantime, I think the sensible thing to do is just introduce a plain old (real) boolean variable in `vars/main.yml`, and selectively set it to `true` when the logfile is defined. This at least spares us an ugly workaround hack.

See e8e796a5bb2f070645912d478de1e7b7494b0b94.